c/meterpreter/source/extensions/stdapi/server/sys/config/config.c

```diff
@@ -93,16 +93,22 @@ DWORD request_sys_config_getenv(Remote *remote, Packet *packet)
 			// grab the value of the variable and stick it in the response.
 			PWCHAR name = met_api->string.utf8_to_wchar(pEnvVarStart);
 			//Ensure we always have > 0 bytes even if env var doesn't exist
-			DWORD envlen = GetEnvironmentVariableW(name, NULL, 0) + 1;
-			PWCHAR wvalue = (PWCHAR)malloc(envlen * sizeof(WCHAR));
-			GetEnvironmentVariableW(name, wvalue, envlen);
+			DWORD envlen = GetEnvironmentVariableW(name, NULL, 0);
+			if ((envlen == 0) && (GetLastError() == ERROR_ENVVAR_NOT_FOUND)) 

```

Comment:
```suggestion
			if (envlen == 0 && GetLastError() == ERROR_ENVVAR_NOT_FOUND) 

```

Reply:
We're being good citizens and directly comparing to `ERROR_ENVVAR_NOT_FOUND`; But I wonder if we should handle length 0 and the potential of `GetLastError()` being some other value? 🤔 

The docs don't suggest that is a possibility:

> If the function fails, the return value is zero. If the specified environment variable was not found in the environment block, [GetLastError](https://learn.microsoft.com/en-us/windows/desktop/api/errhandlingapi/nf-errhandlingapi-getlasterror) returns `ERROR_ENVVAR_NOT_FOUND`.
> https://learn.microsoft.com/en-us/windows/win32/api/processenv/nf-processenv-getenvironmentvariablew

But if it _did_ happen, we'd end up sending back mystery bytes to msfconsole again

Reply:
Seems like if `envlen` is 0 and `GetLastError()` returns something other than `ERROR_SUCCESS` we should take this path.

Reply:
Sounds like a good call, I'll adjust

Reply:
@smcintyre-r7 We found that you can't have a 0 length env var - even when you directly modify the registry - so probably skipping the `GetLastError()` check entirely seems cromulent?

```suggestion
			if (envlen == 0)

```

Calling `GetLastError` could maybe be useful for a different `dprintf` log, but it feels like we should be able to skip it here